Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -506,8 +521,9 @@ public final class ManifestLoader: ManifestLoaderProtocol {
         identityResolver: IdentityResolver,
         delegateQueue: DispatchQueue,
         fileSystem: FileSystem,
-        observabilityScope: ObservabilityScope
-    ) throws -> ManifestJSONParser.Result {
+        observabilityScope: ObservabilityScope,
+        completion: @escaping (Result<ManifestJSONParser.Result, Error>) -> Void
+    ) throws {
```

Comment:
functions that take a completion handler with a result type should not also be throwing as it mixes two error handling patterns that are easy to get wrong

Reply:
I like it.